### PR TITLE
Add solr helm terraform deployment

### PR DESCRIPTION
**Pull request by mckornfield**

### CODE REVIEW

File: solr/solr_chart/main.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  namespace        = "solr"
+  create_namespace = true
+
+
+  values = [
+    "${file("${path.module}/values.yaml")}",
+    var.values_file
+  ]
+
+}
```
Comment (dadiorchen) [inline solr/solr_chart/main.tf:29]: @mckornfield I'm trying to understand what you are doing, let me know if I was wrong:
1. We are using Terraform's helm provider to install Solr to K8S
2. What are the benefits of using Terraform  rather then use ansible?
3. What's the things need to save to Terraform states?
4. I can not find the code to set the namespace for the Solr?

Comment (mckornfield) [reply solr/solr_chart/main.tf:29]: 1. Yep that's right
2. To be honest, it's about the same, except that there's a state reference somewhere. I've recently come to use terraform for cluster installations that are more permanent, and I think I like it more than ansible, as far as being able to "show" what the current state is, and detect drift
3. That script I added basically pulls keys from a locally filled out `~/.netrc` to query the appropriate keys
4. https://github.com/Greenstand/treetracker-infrastructure/pull/165/files#diff-24ac15345fd7e132af56f8f88808b1d8196f8ad1500cbbd46ce36851b51cfca3R20 this line sets the namespace

Comment (dadiorchen) [reply solr/solr_chart/main.tf:29]: Thank you for you great answer!

**Review by dadiorchen — APPROVED** [review]

**Comment by dadiorchen** [comment]
@mckornfield I tried to use the Solr on dev env, seems we need to do something else to make the service available, in the namespace `solr` there are two pod: 

```
pod/solr-operator-74fff58468-g4zpb                      1/1     Running   3 (10h ago)   6d4h
pod/solr-operator-zookeeper-operator-76595c9658-8lgn8   1/1     Running   0             6d4h
```

They seem to be some controller of the Solr?  According to the doc below, we need to use the `operator` to create `Solr cluster` ?

```
Solr Operator - A management layer that runs independently in Kubernetes. Only deploy 1 per Kubernetes cluster or namespace.
Solr - A SolrCloud cluster. In order to deploy this Helm chart successfully, you must first install the Solr Operator and Solr CRDs. Follow the instructions on the ArtifactHub page below.
The Solr Operator supports creating and managing Solr Clouds.
```

https://solr.apache.org/operator/resources.html#helm-charts

https://apache.github.io/solr-operator/docs/solr-cloud/

**Comment by mckornfield** [comment]
yep sounds like it needs a custom resource to define a solr cluster, which I can include in this

**Comment by mckornfield** [comment]
OK got a push for ya. I applied it in dev and test, let me know if it looks good and I can add it to prod as well. If you want node selectors etc. then let me know too

**Comment by mckornfield** [comment]
@dadiorchen addressed, though I worry the resourcing might be too big in dev? Let me know if I need to turn it down

**Comment by dadiorchen** [comment]
@mckornfield thanks! 
well, our data to dump into the Solr is pretty limited, and the workload would not be huge, if you can easily reduce it, that's great, but if not, no problem.

Now I can visit the dashboard by post-forword the service to local:
image

I need to read some doc to find out how to set up the user system so I can safely expose it.

**Comment by dadiorchen** [comment]
Yea, I will let you know if it is ready for prod, before it happens, I still have something to do:
1. #167 
2. The developer needs to dump the data into Solr to test the feature and dev the UI on the app.

**Pull request merged**